### PR TITLE
Gh 2052/wallet connection vc

### PR DESCRIPTION
**Pull request by DmitryBespalov**
Handles #2052 

Changes proposed in this pull request:
- Adapt and reuse parts from the existing handling of connection
- Incidentally, this also implements handling of disconnection and updating a connection
- So far, MetaMask connection worked OK.

### CODE REVIEW

File: Multisig.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -262,15 +262,6 @@
           "version": "2.0.1"
         }
       },
-      {
```
Comment (MouazAlzahabi) [inline Multisig.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: Do you want to return this back after updating the library?

File: Multisig/Features/Connect to Wallet/Logic/Models/WCAppRegistryEntry.swift [hunk 2]
```diff
@@ -77,10 +77,77 @@ class WCAppRegistryEntry {
...
+        } else {
+            // fallback to the connection url
+            return URL(string: url.absoluteString)
+        }
+    }
+
+    /// Link to switch to the wallet
+    ///
+    /// see: https://docs.walletconnect.com/mobile-linking#for-ios
+    func navigateLink(from url: WebConnectionURL) -> URL? {
```
Comment (MouazAlzahabi) [inline Multisig/Features/Connect to Wallet/Logic/Models/WCAppRegistryEntry.swift:140]: Is this one used?

File: Multisig/Features/Connect to Wallet/UI/WalletConnectionViewController.swift [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            onCancel()
+        }
+    }
+    
+    func didUpdate(connection: WebConnection) {
+        switch connection.status {
+        case .opened:
+            onSuccess(connection)
+        case .final:
+            // show failed to connect, close screen
```
Comment (MouazAlzahabi) [inline Multisig/Features/Connect to Wallet/UI/WalletConnectionViewController.swift]: Is this comment still valid ?

**Review by MouazAlzahabi — APPROVED** [review]

File: Multisig/Features/Connect to Wallet/UI/WalletConnectionViewController.swift [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    }
+    
+    deinit {
+        WebConnectionController.shared.detach(observer: self)
+    }
+
+    override func willMove(toParent parent: UIViewController?) {
+        super.willMove(toParent: parent)
+        // needed to react to the "swipe down" to close the modal screen
+        parent?.presentationController?.delegate = self
```
Comment (elgatovital) [inline Multisig/Features/Connect to Wallet/UI/WalletConnectionViewController.swift:93]: this looks hacky :) Is this needed for presentationControllerWillDismiss to be triggered?

File: Multisig.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (DmitryBespalov) [reply Multisig.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: hm,  yes, somehow this was deleted

File: Multisig/Features/Connect to Wallet/Logic/Models/WCAppRegistryEntry.swift (hunk 2 above)
Comment (DmitryBespalov) [reply Multisig/Features/Connect to Wallet/Logic/Models/WCAppRegistryEntry.swift:140]: it should be used later for signature and transaction requests

File: Multisig/Features/Connect to Wallet/UI/WalletConnectionViewController.swift (hunk 3 above)
Comment (DmitryBespalov) [reply Multisig/Features/Connect to Wallet/UI/WalletConnectionViewController.swift]: no, not valid anymore

**Review by elgatovital — APPROVED** [review]

**Pull request merged**